### ISSUE TITLE
Eliminate witness votes after some time

### ISSUE THREAD

**Issue by Nemo1369** [GolosChain/golos#99]
Witness votes elimination after some time is required to automatically downvote inactive witness.

**Comment by Gandalf-the-Grey** [comment]
You mean automatic un-approval of witnesses after certain time following `shutdown_witness`?

**Comment by serejandmyself** [comment]
It means that after 3 months a vote of user X will automatically get revoked. This was proposed by me long time ago on steemit, but was not upheld by the witnesses back then. Currently the situation with the witnesses is atrocious. People aren't doing anything (apart from 10-15%)

**Comment by Gandalf-the-Grey** [comment]
Yes, you are right, situation is terrible, but I'm not sure if solving it this way is a good idea. I'm one of those witnesses that were against it on Steem.

I can't fully assess situation on Golos because of my language barrier, but as far as I can see, there are a lot of inactive witnesses holding high positions, but IMHO that's **NOT** their fault. Stakeholders are responsible for this situation.

`shutdown_witness` mechanism is enough to keep failing nodes (not up to date, not in sync, etc.) out of queue without impacting participation rate.

Another issue are witnesses that are not really active on the platform, but I hold the position that all approvals and un-approvals should be done consciously by stakeholders. Doing this automatically might impact network reliability and security.

I'll try to improve the situation without such drastic step, i.e. by [convincing people to vote consciously](https://golos.io/en/@gtg/golos-witness-witnessing-witnesses-use-your-own-voice-wisely)

I hope that would help....

**Comment by roelandp** [comment]
I am totally with @Gandalf-the-Grey's opinion here, if you want witnesses to support the network and keep Golos chain alive and in good health with 100% participation rate, it is not smart to auto-unvote witnesses. Maybe I am biased because I am a witness, but imagine all witnesses being removed after 3 months after being voted in it would mean the top 17 or so would instantly be removed and have 0 votes. This would mean the network instantly depends (and counts on) the following 17 in the list to start producing. If you look at that list from 20 -> onwards to 37 you see many missed blocks (albeit not being fulltime producer). IAW it is a big gamble to depend on 2nd degree witnesses. 

It is also not unfair imho the witness pay in golos is very reasonable, with about 0.11 GOLOS Power per produced block. 

I think it would be smart to stimulate people to actively 'fire' witnesses if they are not performing their duties. 

Besides, with having an auto-removal tool, would definitely lead to someone creating an 'auto re-vote'-voting-ring tool to revote witnesses as soon as they are auto-voted out, I think.

**Comment by serejandmyself** [comment]
Is this done for witnesses?

**Comment by roelandp** [comment]
I guess not as it would than have a commit referencing this issue. My opinion on this matter still stands. U want people to invest in hardware to support a growing network and not have people at random times starting the "ask for votes" stuff again. It would be good to auto unvote / disable witnesses after x amount blocks missed but that would be to serve the network / participation rate.

**Comment by jackvote** [comment]
downvote inactive - I understand this as removing votes some time after the last block signed. If a witness hasn't signed the blocks for six months, why should he be on the list?
The time before the vote is taken can be voted by witnesses.

Я понимаю это как удаление голосов через некоторое время после последнего подписанного блока. Если делегат шесть месяцев не подписывал блоки, почему он должен быть в списке?
Время до снятия голосов может быть голосуемо делегатами.
